src/zjs_gfx.c

```diff
@@ -0,0 +1,468 @@
+    gfx_data_t argData;
+    args_to_data(&argData, argc, argv);
+    return zjs_gfx_draw_char_priv(argData.coords[0], argData.coords[1], argData.text[0], argData.color, argData.size, handle);
+}
+
+static ZJS_DECL_FUNC(zjs_gfx_draw_string) //(x, y, c, color, bg, size) {
+{
+    // requires: Requires 4 arguments
+    //           arg[0] - x coord of the top left.
+    //           arg[1] - y coord of the top left.
+    //           arg[2] - text.
+    //           arg[3] - color array.
+    //           arg[4] - optional size of character.
+    //
+    //  effects: Draws a character on the screen
+
+    ZJS_VALIDATE_ARGS(Z_NUMBER, Z_NUMBER, Z_STRING, Z_ARRAY, Z_OPTIONAL Z_NUMBER);
+    ZJS_GET_HANDLE(this, gfx_handle_t, handle, gfx_type_info);
+    gfx_data_t argData;
+    args_to_data(&argData, argc, argv);
+    jerry_value_t ret = ZJS_UNDEFINED;
+
+    for (u8_t i = 0; i < argData.textSize; i++) {
+        ret = zjs_gfx_draw_char_priv(argData.coords[0] + (i * 6 * argData.size), argData.coords[1], argData.text[i], argData.color, argData.size, handle);
+        if (jerry_value_has_error_flag (ret)) {
+            return ret;
+        }
+    }
+    return ret;
+}
+
+static ZJS_DECL_FUNC(zjs_gfx_set_cb)
+{
+    // requires: Requires 4 arguments
+    //           arg[0] - The width of the screen.
+    //           arg[1] - The hight of the screen.
+    //           arg[2] - The init callback to use.
+    //           arg[3] - The drawRect callback to use.
+    //           arg[4] - optional JS 'this' object
+    //
+    //  effects: Initializes the GFX module
+
+    ZJS_VALIDATE_ARGS(Z_NUMBER, Z_NUMBER, Z_FUNCTION, Z_FUNCTION, Z_OPTIONAL);
+
+    gfx_handle_t *handle = zjs_malloc(sizeof(gfx_handle_t));
+
+    if (!handle) {
+        return ZJS_ERROR("could not allocate handle\n");
+    }
+
+    handle->screenInitCB = jerry_acquire_value(argv[2]);
+    handle->drawDataCB = jerry_acquire_value(argv[3]);
+    handle->jsThis = jerry_acquire_value(argv[4]);
+
+    jerry_value_t gfx_obj = zjs_create_object();
+    jerry_set_prototype(gfx_obj, zjs_gfx_prototype);
+    jerry_set_object_native_pointer(gfx_obj, handle, &gfx_type_info);
+    jerry_call_function(handle->screenInitCB, handle->jsThis, NULL, 0);
+    return gfx_obj;
+}
+
+jerry_value_t zjs_gfx_init()
+{
+    zjs_native_func_t proto[] = {
+        { zjs_gfx_fill_rect, "fillRect" },
+        { zjs_gfx_draw_pixel, "drawPixel" },
+        { zjs_gfx_draw_line, "drawLine" },
+        { zjs_gfx_draw_v_line, "drawVLine" },
+        { zjs_gfx_draw_h_line, "drawHLine" },
+        { zjs_gfx_draw_rect, "drawRect" },
+        { zjs_gfx_draw_char, "drawChar" },
+        { zjs_gfx_draw_string, "drawString" },
+        { NULL, NULL }
+    };
+
+    zjs_gfx_prototype = zjs_create_object();
+    zjs_obj_add_functions(zjs_gfx_prototype, proto);
+
+    jerry_value_t gfx_obj = zjs_create_object();
+    //zjs_obj_add_function(gfx_obj, zjs_gfx_set_cb, "init");
```

Comment:
there are a few of these stray things left commented out

Reply:
Whoops thanks I'll clean that up

Reply:
Fixed